### DISCUSSION TITLE
enable mtk7921 to connect to u-nii bands 6,7 and 8

### DISCUSSION THREAD

**Discussion by konikofi** [WLAN-Pi/feedback#53] [Hardware - WLAN Pi R4]
The MTK7921 NIC currently has a problem connecting to U-NII bands 6,7 and 8 which is 6GHz channels 97-233.

wpa_supplicant has a feature to overcome this problem called freq_list
however, the drawback is that this directive will limit the mtk7921 to only scan the frequencies in the freq_list, but it will allow it to connect to those upper 6GHz channels that it can't seem to do by default yet

add the following lines to your Pi R4 /etc/wpa_supplicant/wpa_supplicant.conf with your SSID and PSK
you can add or remove frequencies to the freq_list as needed

```
sae_pwe=2
freq_list=6775 6855 6935

network={
  ssid="asus11axe6"
  psk="hello123"
  priority=10
  key_mgmt=SAE
  ieee80211w=2
}
```
then start with:
```wpa_supplicant -c /etc/wpa_supplicant/wpa_supplicant.conf -i wlan0 &```
which should show a connection event
```
root@wlanpi-c0c:~# Successfully initialized wpa_supplicant
wlan0: SME: Trying to authenticate with f0:2f:74:7c:8d:b8 (SSID='asus11axe6' freq=6775 MHz)
wlan0: SME: Trying to authenticate with f0:2f:74:7c:8d:b8 (SSID='asus11axe6' freq=6775 MHz)
wlan0: PMKSA-CACHE-ADDED f0:2f:74:7c:8d:b8 0
wlan0: Trying to associate with f0:2f:74:7c:8d:b8 (SSID='asus11axe6' freq=6775 MHz)
wlan0: Associated with f0:2f:74:7c:8d:b8
wlan0: CTRL-EVENT-SUBNET-STATUS-UPDATE status=0
wlan0: CTRL-EVENT-REGDOM-CHANGE init=COUNTRY_IE type=COUNTRY alpha2=US
wlan0: WPA: Key negotiation completed with f0:2f:74:7c:8d:b8 [PTK=CCMP GTK=CCMP]
wlan0: CTRL-EVENT-CONNECTED - Connection to f0:2f:74:7c:8d:b8 completed [id=0 id_str=]
```
verified on wlanpi os 3.1.2 using R4 model with comfast 951-ax usb nic
```Linux wlanpi-c0c 6.1.9-v8-wlanpi+ #1 SMP PREEMPT Thu Feb  2 00:29:25 UTC 2023 aarch64 GNU/Linux```

**Comment by jiribrejcha** [comment]
Thanks for sharing this workaround 👍🏻 We’ve been hoping for one if the upcoming firmware releases to fix this, but this issue is still present on MT7921K and CF-951AX.
